src/color/creating_reading.js

```diff
@@ -1220,8 +1237,16 @@ function creatingReading(p5, fn){
    * </div>
    */
   fn.saturation = function(c) {
-    // p5._validateParameters('saturation', arguments);
-    return this.color(c)._getSaturation();
+    const colorMode = (
+      this._renderer.states.colorMode === HSB ||
```

Comment:
Minor/optional: isn't this the same as `(this._renderer.states.colorMode === HSB) ? HSB : HSL`? that seems a little more readable.